overlord/ifacestate/handlers.go

```diff
@@ -552,6 +630,100 @@ func (m *InterfaceManager) defaultContentProviders(snapName string) map[string]b
 	return defaultProviders
 }
 
+func checkAutoconnectConflicts(st *state.State, plugSnap, slotSnap string) error {
+	for _, task := range st.Tasks() {
```

Comment:
The body of this function and the body of checkDisconnectConflicts look almost identical. Could we extract them into a common helper and just pass the affectingTaskKinds into it? AFAICT this is the only difference.

Reply:
they were one function before the last pass, but there are subtle differences, Gustavo input was that they were too subtle to make for a readable single function

Reply:
Totally fine to keep it this way if it evolved like this. Sorry that I misssed the history here.